crates/bevy_ecs/src/event.rs

```diff
@@ -798,48 +810,77 @@ impl<'a, E: Event> ExactSizeIterator for EventIteratorWithId<'a, E> {
     }
 }
 
+/// A registry of all of the [`Events`] in the [`World`], used by [`event_update_system`]
+/// to update all of the events.
 #[doc(hidden)]
 #[derive(Resource, Default)]
-pub struct EventUpdateSignal(bool);
+pub struct EventRegistry {
+    needs_update: bool,
+    // SAFETY: The ID and the function must be used in conjunction, or improper type casts
```

Comment:
"in conjunction" is inadequately clear for a safety comment IMO.

I think that this is something closer to "the ComponentId corresponding to the resource must correspond to the erased type of the fn pointer"

Reply:
Tried rewording it, check again?